File: ios/Classes/FlutterLlamaCppDartLibPlugin.m
#import "FlutterLlamaCppDartLibPlugin.h"

@implementation FlutterLlamaCppDartLibPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_llama_cpp_dart_lib"
            binaryMessenger:[registrar messenger]];
  FlutterLlamaCppDartLibPlugin* instance = [[FlutterLlamaCppDartLibPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: macos/Classes/FlutterLlamaCppDartLibPlugin.swift
import Cocoa
import FlutterMacOS

public class FlutterLlamaCppDartLibPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "flutter_llama_cpp_dart_lib", binaryMessenger: registrar.messenger)
    let instance = FlutterLlamaCppDartLibPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
